### PR TITLE
Added functionality when moving an issue to a Resolved/Closed state

### LINKED ISSUE #64: Add a reason when closed or resolved
When an issue is moved to Closed or Resolved state a reason should be added to the issue so that the UI crosses off the issue.

If a Closed or Resolved issue is moved back to any other state then remove the reason.

### PR DESCRIPTION
**Pull request by sfradkin**
Adding functionality to require a resolution when closing an issue, also can take optional comment when closing, updates readme

fixes #64

### CODE REVIEW

File: fc/cli/task.py [hunk 1]
```diff
@@ -31,7 +39,7 @@ def move(username: str, task_id: str, state: str):
         cli_library.fail_execution(1, 'Task search failed with {}'.format(exception))
 
     if the_task is not None:
-        the_task.transition(state)
+        the_task.transition(state, the_resolution, comment)
```
Comment (halprin) [inline fc/cli/task.py]: Does JIRA allow setting the resolution or comment on a transition other than Resolved or Closed?  What happens?

File: fc/cli/task.py [hunk 2]
```diff
@@ -17,8 +17,16 @@ def task():
...
-    cli_library.echo('Transitioning task')
+@click.option('--resolution')
+@click.option('--comment')
+def move(username: str, task_id: str, state: str, resolution: str, comment: str):
+
+    the_resolution = resolution
+
+    if state in ['Resolved', 'Closed']:
+        if resolution is None:
+            the_resolution = cli_library.prompt('Please enter a resolution',
```
Comment (halprin) [inline fc/cli/task.py]: What was your reasoning behind having the extra `the_resolution` variable in `task.py` in the FCLI PR?

File: fc/cli/task.py (hunk 1 above)
Comment (sfradkin) [reply fc/cli/task.py]: comments are ignored.  I believe adding a resolution when you don't need one results in a 400 error.

**Review by halprin — APPROVED** [review]
Looks good to me.  Looks like there's a PEP8 error though since CI failed.

**Pull request merged**